mono/metadata/threads.c

```diff
@@ -6665,8 +6647,6 @@ mono_threads_summarize (MonoContext *ctx, gchar **out, MonoStackHash *hashes, gb
 				g_assert (mem);
 				success = mono_threads_summarize_execute_internal (ctx, out, hashes, silent, mem, provided_size, TRUE);
 				summarizer_supervisor_end (&synch);
-			} else {
-				summarizer_supervisor_wait (&synch);
```

Comment:
I think this is the child process - ie this is where the `exec` would need to happen.

Reply:
Yup I got it reversed - though the patch will still end up not having `summarizer_supervisor_wait` and the change will be in `summarizer_supervisor_start`